### DISCUSSION TITLE
Problemas con CheckStyle

### DISCUSSION THREAD

**Discussion by MariaPaulaBaldivia** [dds-jv/foro#198] [Consultas de herramientas] [answered]
Buenas! 

Tengo un problema con el Chechstyle. Me resalta todos mis archivos de esta manera
![image](https://user-images.githubusercontent.com/62667460/116120687-31752c80-a696-11eb-8533-c638de4306e5.png)

Creería que es un problema en la configuración, pero lo instalé siguiendo el readme.
(El tabulado lo mando por las dudas)
![image](https://user-images.githubusercontent.com/62667460/116120530-fffc6100-a695-11eb-89e8-4f9bb1f8ab62.png)
![image](https://user-images.githubusercontent.com/62667460/116121167-bc562700-a696-11eb-831b-465480973722.png)

Cuando realizo el comando mvn verify, me muestra esto
![image](https://user-images.githubusercontent.com/62667460/116121878-73eb3900-a697-11eb-919d-ac0253a8c2d7.png)
No me tira ningún error pero tampoco me muestra la cobertura del proyecto.

Tal vez no es un problema en sí, pero como no conozco el entorno preferí preguntar antes de entregar el proyecto.

Gracias de antemano.

**Comment by nanderson87** [comment, marked as answer]
Hola María:

Estando dentro de una clase que te resalte, por ejemplo `Prenda.java` podes tocar el menú problems (o problemas) que se encuentra abajo a la izquierda para ver un poco mejor porque te marca esas líneas.

![image](https://user-images.githubusercontent.com/11875266/116125554-068dd700-a69c-11eb-9c1c-fb3bae085ba3.png)

Otra forma de llegar a lo misma es dentro de la clase el icono de warning ⚠️ que aparece arriba a la derecha. En tu caso ese que tiene el valor 19 para la clase `Prenda.java`

![image](https://user-images.githubusercontent.com/11875266/116125773-494faf00-a69c-11eb-8b26-4580ac889904.png)

Dentro de ese menú te va explicar el motivo por el cual te esta dando ese warning y sugerencias del IDE para tratar el mismo. Fijate si eso te sirve, en todo caso nos podes pasar una captura para verlo un poco mejor. 

Otra opción para ver las sugerencias del IDE es poner el cursor sobre la línea que te marca y debería salirte una lamparita color amarilla o una nueva ventana diciéndote porque marco la misma y dandote posibles sugerencias.

Basicamente son varias formas de llegar a lo mismo. Espero que te sirva

**Comment by MariaPaulaBaldivia** [comment]
Hola Nicolas! 

Mirá esto es lo que me tira Prenda.java (el que envié anteriormente)

![image](https://user-images.githubusercontent.com/62667460/116146522-f6cebc80-a6b4-11eb-9224-44ea24336d3e.png)

Es algo de CheckStyle, pero no sé como arreglarlo

**Reply by nanderson87** [reply]
Ok. Las que nos van a interesar son la indentación. Las que son del estilo:

`Checkstyle ... el nivel de sangrado X no está al nivel correcta, N`

Si te fijas los niveles correctos que te propone son los que configuraste en` Settings >> CodeStyle >> Java >> Tabs and idents`.  

Es posible que hayas configurado esto posteriormente de los tabs después de haber creado estas clases que te marca?

En fin, podrías ir solucionandolo uno a uno, seleccionando un registro de estos y viendo que te propone "la lamparita" o bien darle formato a todo el archivo.  Para hacer esto ultimo parada dentro de una clase, abrií el format file.

For Windows : Ctrl + Alt + L
For Ubuntu : Ctrl + Alt + Windows + L
For Mac : ⌥ (Option) + ⌘ (Command) + L

![image](https://user-images.githubusercontent.com/11875266/116149795-f46e6180-a6b8-11eb-9dca-9894a28ea78a.png)

Te debería salir algo asi. Seleccionamos Whole file (o todo el archivo, dependiendo el idioma del IDE) y esto debería acomodarte las espacios correctamente ya que por la imagen que pasaste lo tenemos configurado bien (tab size:2; indent:2; c. indent:4)

**Comment by MariaPaulaBaldivia** [comment]
Gracias!! Lo puede solucionar haciendo Ctrl + Alt + L   en cada archivo.

Una última consulta, al hacer mvn verify me sigue mostrando esto
![image](https://user-images.githubusercontent.com/62667460/116155816-06540280-a6c1-11eb-904f-4beec3482d74.png)

Además no me muestra la cobertura para poder asegurarme que cumple con el porcentaje mínimo que se pide. 

¿Estaría mal que entregue el proyecto con esos warnings? 

Mil disculpas si estoy siendo muy reiterativa

**Reply by nanderson87** [reply]
No hay problema :smile:

- Respecto a esto estos warnings

![image](https://user-images.githubusercontent.com/11875266/116167728-37d7c880-a6d7-11eb-970a-2a6b9235f5bf.png)


Lo podes omitir totalmente, no tiene nada que ver con tu configuración, es la versión que esta usando `maven` de checkstyle. Y nada, es solo un warning que nos esta avisnando que hay una versión más actual. Lo importante es esto:

![image](https://user-images.githubusercontent.com/11875266/116168061-fc89c980-a6d7-11eb-9361-c3191e920307.png)


Si algunas de las verificaciones no pasan al ejecutar el comando `mvn verify` lo que verías es que el build termina fallando (por no haber pasado alguna de las verificaciones). Verías algo así:


![image](https://user-images.githubusercontent.com/11875266/116168703-6eaede00-a6d9-11eb-96d4-1678fe5672cf.png)

Y algunos mensaje de error avisando que verificación o verificaciones no pasaron. En tu caso el build terminó de forma satisfactoria por lo que tu proyecto esta pasando todas las verificaciones. Por lo tanto, en ese sentido no tenes ninguna restricción para continuar con la entrega.

Como dato de color, si la verificación que no pasa es la de cobertura verías algo similar a esto:

![image](https://user-images.githubusercontent.com/11875266/116168986-0ca2a880-a6da-11eb-9fbd-760a6c31a74a.png)

**Reply by MariaPaulaBaldivia** [reply]
Entonces todo resuelto.

Gracias por ayudarme y sacarme las dudas!

Saludos
